qiskit/visualization/matplotlib.py

```diff
@@ -282,14 +283,23 @@ def _get_text_width(self, text, fontsize):
             if num_carets:
                 text = text.replace('^', '', num_carets)
 
+            # This removes some arbitrary \n's and spaces that are returned
+            # by latex_to_text when parameters contain pi. Not sure why this
+            # happens and may change in a future release of latex_to_text. It
+            # also changes hyphen to + to match width of math mode minus sign.
+            if param:
+                text = text.replace('\n    ', '')
+                text = text.replace('\n', '')
```

Comment:
```suggestion
                text = text.rstrip()
```

Reply:
Actually rstrip doesn't work. If you look at #4845, you'll see the 10's and 32's are mixed in the string.